code/game/machinery/cloning.dm

```diff
@@ -308,7 +308,7 @@
 		return
 
 	if((occupant) && (occupant.loc == src))
-		if((occupant.stat == DEAD) || (occupant.suiciding) || !occupant.key)  //Autoeject corpses and suiciding dudes.
+		if((occupant.stat == DEAD) || (occupant.suiciding) || !occupant.key || occupant.hellbound)  //Autoeject corpses and suiciding dudes.
 			locked = 0

```

Comment:
If someone gets scanned before they sign a contract, their data is still on file in spite of the soul being captured - you should still add the check in `growclone`